api-get-options.go

```diff
@@ -44,7 +44,7 @@ func (o GetObjectOptions) Header() http.Header {
 	for k, v := range o.headers {
 		headers.Set(k, v)
 	}
-	if o.ServerSideEncryption != nil && o.ServerSideEncryption.Type() != encrypt.S3 {
```

Comment:
Maybe this should be `if o.ServerSideEncryption != nil && o.ServerSideEncryption.Type() == encrypt.SSEC {`

Reply:
I'm okay with both solutions. `o.ServerSideEncryption != nil` should cause the server to fail and the additional check `o.ServerSideEncryption.Type() == encrypt.SSEC` just drops the parameter. Whatever a typical user would prefer.

Reply:
that's a good suggestion, done.